### PR TITLE
Add node-midi (just 'midi' on npm) type declarations

### PR DESCRIPTION
**Pull request by mattsoulanille**
Add type declarations for [node-midi](https://github.com/justinlatimer/node-midi), which is published to npm as [midi](https://www.npmjs.com/package/midi).

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/midi/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
+// Type definitions for midi 2.0
+// Project: https://github.com/justinlatimer/node-midi (Does not have to be to GitHub, but prefer linking to a source code repository rather than to a project website.)
```
Comment (peterblazejewicz) [inline types/midi/index.d.ts:2]:
```suggestion
// Project: https://github.com/justinlatimer/node-midi
```

File: types/midi/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    /** Send a MIDI message */
+    send(message: MidiMessage): void;
+    /** Send a MIDI message */
+    sendMessage(message: MidiMessage): void;
+}
+
+/** @deprecated */
+export const input: { new (): Input };
+/** @deprecated */
+export const output: { new (): Output };
```
Comment (peterblazejewicz) [inline types/midi/index.d.ts:71]:
```suggestion
export const input: typeof Input;
/** @deprecated */
export const output: typeof Output;
```
IMO this notation of the same construct (a class type) is more widely used

**Comment by mattsoulanille** [comment]
Thanks for the review, @peterblazejewicz! I've accepted your suggestions.

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@mattsoulanille thanks!

**Comment by mattsoulanille** [comment]
Ready to merge

**Pull request merged**